Dockerfile

```
@@ -0,0 +1,7 @@
+FROM python:3.10.9
+WORKDIR /root/
+RUN apt-get update && apt-get upgrade -y
+ENV PATH="/root/.local/bin:${PATH}"
+COPY requirements.txt .
+RUN pip3 install --user --upgrade pip
+RUN pip3 install --user -r requirements.txt
```

Combine multiple RUN commands into a single one. This reduces the number of image layers and improves build speed.

no-cache-dir avoid caching the downloaded packages.

```suggestion
FROM python:3.10.9

WORKDIR /root/
ENV PATH="/root/.local/bin:${PATH}"

COPY requirements.txt .

RUN apt-get update \
    && apt-get upgrade -y \
    && pip install --upgrade pip \
    && pip install --no-cache-dir -r requirements.txt
```